### ISSUE TITLE
Restore purchases without entering password

### ISSUE THREAD

**Issue by mattyat3s** [j3k0/cordova-plugin-purchase#136]
Hi, got my app working great with the plugin, there's just one weird issue.

If I click my "Restore Purchase" button which fires the "store.refresh();" I get the Apple dialog asking for my password but in the background I can see my payment buttons go from Buy to Download.

I have the code to change the buttons in the when approved event, which seems to be approving before the password.

I've tried the code in the owned and updated events but they also load before the password.

It works perfectly when purchasing, the above is just when firing the store.refresh() a second time.

Is there a callback I should be waiting for?

Thanks,
Matt

**Comment by mattyat3s** [comment]
Just tested on iOS 6 and it works fine, it's just iOS 8 that i'm having the issue above with.

**Comment by filR** [comment]
I am getting the password prompt on iOS 6 whenever I call `store.refresh()` a second time.

And I am also getting all callbacks before needing to enter the password.

**Comment by OJcode14** [comment]
I also experiencing this issue.

**Comment by Heidschnucke** [comment]
I have the same problem.

**Comment by Heidschnucke** [comment]
I think, I solved it for me after weeks of desperation. Perhaps your problems have the same origin.

The unnecessary login prompts are not a bug in this plugin. It is caused by using and testing with test users. If you buy something with a test user and do not finish the transaction, the app store will always try to finish the transaction in future until it is finished. If you haven't finish transactions on different test users, the prompt will even fire for every one of them. 

I found some suggestions under:
http://stackoverflow.com/questions/25735474/in-app-purchasing-password-prompt

But the only thing that helped was resetting the device (an iPad in my case).
